### ISSUE TITLE
[Help]: regression trend removal causing artifacts

### ISSUE THREAD

**Issue by SteffanDavies** [mobigroup/gmtsar#98]
Regression method is causing border artifacts:

![image](https://github.com/AlexeyPechnikov/pygmtsar/assets/50405969/4daf61e7-8050-48f5-96c3-7fe580be1ed1)

![image](https://github.com/AlexeyPechnikov/pygmtsar/assets/50405969/c116d731-8a99-431d-95d1-f5e9f73fdaed)

Do you recommend any workaround except for increasing chunksize (it is currently 2048)

**Comment by AlexeyPechnikov** [comment]
You are right: we need to increase the chunksize to have a single chunk for the regression and revert it back later. To prevent performance degradation just filter out low correlation pixels. Typically, 1000 - 10000 pixels are more than enough for robust  regression estimation.

**Comment by OKI-Wu** [comment]
Hi, sorry if I'm not supposed to reply here, but I encountered the same issue. The problem is I'm dealing with a huge area (like 30+ subswaths), so making it a single chunk is probably not practical. Is it possible to fit the regression model with some high coherence pixels then apply the model to each chunk? Or do you have any alternatives? Thank you.

**Comment by AlexeyPechnikov** [comment]
One single chunk means a consistent model, when a set of smaller chunks can be consistent or not. There is no problem with any size of the area because we can vary the resolution - for a resolution like 1 km it can be processed on a ~1GB RAM host. Select a reasonable resolution and produce the interferograms for the regression model, filter out not well coherent pixels, and interpolate the result to your actual resolution. The rule of thumb is to use low resolution SBAS processing to apply spatial operations like regression and unwrapping (we can compute regression without unwrapping at all) as needed and perform PSI processing per pixel later.
